giskard_vision/core/detectors/metadata_scan_detector.py

```diff
@@ -0,0 +1,154 @@
+from typing import Any, Callable, List, Sequence
+
+import numpy as np
+import pandas as pd
+
+from giskard_vision.core.detectors.base import (
+    DetectorVisionBase,
+    IssueGroup,
+    ScanResult,
+)
+from giskard_vision.utils.errors import GiskardImportError
+
+
+class MetadataScanDetector(DetectorVisionBase):
+    """
+    Detector based on Giskard scan that looks for issues based on metadata
+
+    Args:
+        surrogate_function: function
+            Function to transform the output of the model and the ground truth into one value
+            that will be used by the scan
+        metric: function
+            Metric to evaluate the prediction with respect to the ground truth
+        type_task: str
+            Type of the task for the scan, ["regression", "classification"]
+    """
+
+    issue_group = IssueGroup(name="Metadata", description="Slices are found based on metadata")
+
+    def __init__(self, surrogate_function: Callable, metric: Callable, type_task: str) -> None:
+        super().__init__()
+        self.surrogate_function = surrogate_function
+        self.metric = metric
+        self.type_task = type_task
+
+    def get_results(self, model: Any, dataset: Any) -> List[ScanResult]:
+        try:
+            from giskard import Dataset, Model, scan
+        except (ImportError, ModuleNotFoundError) as e:
+            raise GiskardImportError(["giskard"]) from e
+
+        meta = dataset.get_meta(0)
+        list_categories = meta.get_categories()
+        list_metadata = list(meta.get_scannable().keys())
```

Comment:
if you only need the keys, let's make rename `get_scannable` to `get_scannables` that returns keys() instead like `get_categories`